src/mono/mono/metadata/image.c

```diff
@@ -2402,6 +2402,9 @@ mono_image_close_except_pools (MonoImage *image)
 		char *old_name = image->name;
 		image->name = g_strdup_printf ("%s - UNLOADED", old_name);
 		g_free (old_name);
+		char *old_filename = image->filename;
+		image->filename = NULL;
+		g_free (old_filename);
```

Comment:
Why this roundabout way of doing it, and not just `g_free (image->filename); image->filename = NULL` ?

Reply:
I think I was previously printing out the file name? Will change.